local/gugcat/renderer.php

```diff
@@ -177,8 +179,12 @@ public function display_aggregation_tool($rows, $activities) {
             $htmlrows .= html_writer::tag('td', $row->studentno);
             $htmlrows .= html_writer::tag('td', $row->surname);
             $htmlrows .= html_writer::tag('td', $row->forename);
+
             foreach((array) $row->grades as $grade) {
-                $htmlrows .= '<td>'.$grade.((strpos($grade, 'No grade') !== false) ? null : $this->context_actions($row->studentno)).'</td>';
+                foreach ($activities as $act) {
+                    $contextaction = $this->context_actions($row->studentno, $act->id, $courseid);
```

Comment:
Why is it you're passing an activity id on the $ishidden parameter of the context_actions function, as well as the course id?

Reply:
And where is the navigation from the context action (ammend grade)  going to the edit form page?

Reply:
yup I think my code for that was not there also